packages/clerk-js/src/utils/captcha/hcaptcha.ts

```diff
@@ -0,0 +1,122 @@
+///<reference types="@hcaptcha/types"/>
+
+import { loadScript } from '@clerk/shared/loadScript';
+import type { CaptchaWidgetType } from '@clerk/types';
+
+import { CAPTCHA_ELEMENT_ID, CAPTCHA_INVISIBLE_CLASSNAME } from './constants';
+
+async function loadCaptcha(url: string) {
+  if (!window.hcaptcha) {
+    try {
+      await loadScript(url, { defer: true });
+    } catch {
+      // Rethrow with specific message
+      console.error('Clerk: Failed to load the CAPTCHA script from the URL: ', url);
+      throw {
+        captchaError: 'captcha_script_failed_to_load',
+      };
+    }
+  }
+  return window.hcaptcha;
+}
+
+export const getHCaptchaToken = async (captchaOptions: {
+  siteKey: string;
+  scriptUrl: string;
+  widgetType: CaptchaWidgetType;
+  invisibleSiteKey: string;
+}) => {
+  const { siteKey, scriptUrl, widgetType, invisibleSiteKey } = captchaOptions;
+  let captchaToken = '',
+    id = '';
+  let isInvisibleWidget = !widgetType || widgetType === 'invisible';
+  let hCaptchaSiteKey = siteKey;
+
+  let widgetDiv: HTMLElement | null = null;
+
+  const createInvisibleDOMElement = () => {
+    const div = document.createElement('div');
+    div.id = CAPTCHA_INVISIBLE_CLASSNAME;
+    document.body.appendChild(div);
+    return div;
+  };
+
+  const captcha: HCaptcha = await loadCaptcha(scriptUrl);
+  let retries = 0;
+  const errorCodes: (string | number)[] = [];
+
+  const handleCaptchaTokenGeneration = (): Promise<[string, string]> => {
```

Comment:
❓  Can we extract parts from `handleCaptchaTokenGeneration` to avoid duplicate code?

Reply:
These duplicated code fragments are temprary until we decide on the provider and will be removed soon. If we decide to provide multiple captcha providers, I will refactor the code, as you suggested!